odoo/addons/base/models/res_users.py

```diff
@@ -696,6 +696,7 @@ def create(self, vals_list):
         users = super(Users, self).create(vals_list)
         setting_vals = []
         for user in users:
+            user.partner_id.with_context(mail_auto_subscribe_no_notify=True).user_id = user
```

Comment:
This is not correct. `user_id` is the sales representative for the partner, not the user linked to the partner. For this you have the `res.partner.user_ids` field which is the inverse of `res.users.partner_id`.

Reply:
Yes, don't think we need user to have themselves as sales rep ...